src/uct/tcp/tcp_ep.c

```diff
@@ -205,17 +216,22 @@ unsigned uct_tcp_ep_progress_rx(uct_tcp_ep_t *ep)
     size_t recv_length;
     ssize_t remainder;
 
-    ucs_trace_func("ep=%p", ep);
+    ucs_assertv(ep->length == 0, "ep=%p", ep);
+
+    ep->buf = ucs_mpool_get_inline(&iface->rx_buf_mp);
+    if (ucs_unlikely(!ep->buf)) {
+        return 0;
+    }
 
-    /* Receive next chunk of data */
     recv_length = iface->config.buf_size - ep->length;
     ucs_assertv(recv_length > 0, "ep=%p", ep);
 
-    status = uct_tcp_recv(ep->fd, ep->buf + ep->length, &recv_length);
+    status = uct_tcp_recv(ep->fd, ep->buf, &recv_length);
     if (status != UCS_OK) {
         if (status == UCS_ERR_CANCELED) {
             ucs_debug("tcp_ep %p: remote disconnected", ep);
             uct_tcp_ep_mod_events(ep, 0, EPOLLIN);
+            ucs_mpool_put(ep->buf);
```

Comment:
minor: you use here ucs_mpool_put, but on line 264 used ucs_mpool_put_inline... what about use same call?